src/main/kotlin/frc/robot/subsystems/drive/AlignCommand.kt

```diff
@@ -64,34 +62,27 @@ val controller =
  */
 fun alignToPose(
     goalPose: Pose2d,
-    linearVelocity: LinearVelocity = MetersPerSecond.zero(),
     tolerance: Pose2d = TOLERANCE,
-    poseSupplier: () -> Pose2d = { drive.pose },
     atGoalDebounce: Time = Seconds.of(0.1),
     holonomicController: Pair<TunableHolonomicDriveController, String> =
         Pair(controller, DEFAULT_CONTROLLER_NAME),
 ): Command =
     runOnce({
-            controller.setTolerance(tolerance)
+            setTolerance(tolerance)
+            resetProfiledPID(drive.pose, drive.fieldOrientedSpeeds)
+            setGoal(goalPose)
             Logger.recordOutput(
                 "Alignment/Controllers/CurrentRunningController",
                 holonomicController.second
             )
         })
         .andThen(
-            run({
-                    drive.runVelocity(
-                        holonomicController.first.calculate(
-                            poseSupplier.invoke(),
-                            goalPose,
-                            linearVelocity.`in`(MetersPerSecond),
-                            goalPose.rotation
-                        )
-                    )
-                })
+            run({ drive.runVelocity(getSpeed(drive.pose).invoke()) })
                 .until(
                     Trigger { controller.atReference() }
                         .debounce(atGoalDebounce.`in`(Seconds))
                 )
         )
         .withName("Drive/AlignToPose")
+
+fun alignCommand(pose: Pose2d): Command = drive.defer { alignToPose(pose) }
```

Comment:
1. You don't need "command" in the function's name.
2. If `alignToPose()` isn't supposed to be used by users, it should be `internal`.